### DISCUSSION TITLE
Dynamically pass collection to a page

### DISCUSSION THREAD

**Discussion by powhound4** [11ty/eleventy#2105] [Q&A]
Currently I have a page layout that I am using to display a collection of products on a page:

layouts/collection-layout.liquid

![image](https://user-images.githubusercontent.com/25691905/142514270-4758a9e7-1455-46b2-b16b-6cd3be10f4c2.png)

I have a markdown file selenite/selenite.md: 
---
    layout: ../layouts/collection-layout.liquid
    selectedCollection: 'selenite'
    title: Selenite
---

When I navigate to /selenite the collection-layout.liquid is auto filled with the collection selenite. However creating a markdown file for each product collection that I would like to display seems a bit redundant and doesn't allow for user generated strings to be displayed in the product collection page. 

Is it possible to pass in a value/string when navigating to a page that can be used to access a collection to avoid creating all these markdown files?

Or is there another way that this can be accomplished?

Thanks

**Comment by pdehaan** [comment]
So you're saying the layout never changes, and the only thing that changes is your `selectedCollection` and `title` values?
I wonder if you could have an array of collection name and title pairs in your `_data/` folder and then just use pagination to generate these index pages from the data.

**Reply by pdehaan** [reply]
Yeah, maybe something like this (if I'm understanding the problem correctly):

```liquid
---
# src/category-index.liquid
pagination:
  data: product-collections
  size: 1
  alias: currCollection
permalink: "{{ currCollection.name | slug }}/"
layout: layouts/collection-layout.liquid
---
```

```js
// src/_data/product-collections.js
module.exports = [
  {
    name: "selenite",
    title: "Selenite",
  },
  {
    name: "cats",
    title: "Cats",
  },
  {
    name: "dogs",
    title: "Dogs",
  },
];
```

---

And my boring .eleventy.js and folder structure looks like this:

```js
// .eleventy.js
module.exports = (eleventyConfig) => {
  eleventyConfig.addFilter(
    "money",
    (price = 0, locale = "en-US", currency = "USD") =>
      `${new Intl.NumberFormat(locale, { style: "currency", currency }).format(
        price
      )} ${currency}`
  );

  return {
    dir: {
      input: "src",
      output: "www",
    },
  };
};
```

```sh
tree src/ www/

src/
├── _data/product-collections.js
├── _includes/layouts/collection-layout.liquid
├── category-index.liquid
├── cats/
│   ├── cats.11tydata.js
│   ├── post1.md
│   ├── post2.md
│   └── post3.md
├── dogs/
│   ├── dogs.11tydata.js
│   ├── post1.md
│   ├── post2.md
│   └── post3.md
└── selenite/
    ├── post1.md
    ├── post2.md
    └── post3.md

www/
├── cats/
│   ├── index.html
│   ├── post1/index.html
│   ├── post2/index.html
│   └── post3/index.html
├── dogs/
│   ├── index.html
│   ├── post1/index.html
│   ├── post2/index.html
│   └── post3/index.html
└── selenite/
    ├── index.html
    ├── post1/index.html
    ├── post2/index.html
    └── post3/index.html

18 directories, 26 files
```

**Reply by pdehaan** [reply]
Thats... suboptimal formatting. Pushed to GitHub at https://github.com/pdehaan/11ty-2105

**Comment by powhound4** [comment]
What I ended up doing for now can be found [here](https://www.11ty.dev/docs/quicktips/tag-pages/) buried in the eleventy docs.

**Reply by cfjedimaster** [reply]
Curious - why do you say it's buried? Pagination is clearly called out in the docs and left hand nav.
